addons/sale/views/sale_order_views.xml

```diff
@@ -669,7 +669,7 @@
                             </kanban>
                         </field>
                         <div class="float-end d-flex gap-1 mb-2 ms-1"
-                             name="so_button_below_order_lines">
+                             name="so_button_below_order_lines" invisible="locked">
```

Comment:
I agree that the at least the discount wizard should not available for locked orders, but I don't think hiding the whole section is the best solution as some custom modules may have added buttons there that make sense for locked orders too.
Could you move the `invisible` attribute on the button below instead? (and add it on the promotions button too in `sale_loyalty`) ?

Thanks

Reply:
@Feyensv 
I have made the div named `so_button_below_order_lines` in `sale.order.form.view` **invisible**. As a result, the `sale_loyalty `buttons will also be hidden when the sale order is locked.